src/Narochno.Slack/SlackConfig.cs

```diff
@@ -11,22 +11,27 @@ public class SlackConfig
         public string WebHookUrl { get; set; }
 
         /// <summary>
+        /// The token for non-webhook API methods to use.
+        /// </summary>
+        public string Token { get; set; }
+
+        /// <summary>
         /// A username to post using
         /// Can be overridden in individual messages
         /// </summary>
-        public Optional<string> Username { get; set; }
+        public string Username { get; set; }
 
         /// <summary>
         /// A channel to post to (example: #general)
         /// Can be overridden in individual messages
         /// </summary>
-        public Optional<string> Channel { get; set; }
+        public string Channel { get; set; }
 
         /// <summary>
         /// The emoji to use when posting (example: :ghost:)
         /// Can be overridden in individual messages
         /// </summary>
-        public Optional<string> Emoji { get; set; }
+        public string Emoji { get; set; }

```

Comment:
It was all optional and felt a bit silly making everything optional, but I'm going to re-work this now because only the incoming webhook method uses the options here, and all other endpoints use `Token`.